README.md

```diff
@@ -224,6 +226,63 @@ In accordance with the [Fetch](https://fetch.spec.whatwg.org/#websocket-opening-
 
 User agents need not perform this normalization on the domains in their static lists; user agents may reject static lists that include non-HTTPS domains.
 
+# Clearing Site Data on Set Transitions
+Sites can change which First-Party Set they are a member of. We need to pay attention to these transitions so that they don’t link user identities across all the FPSs they’ve historically been in. In particular, we must ensure that a domain cannot transfer a user identifier from one First-Party Set to another when it changes its set membership.
+
+In order to achieve this, site data needs to be cleared on certain transitions. The clearing should behave like [`Clear-Site-Data: "*"`](https://www.w3.org/TR/clear-site-data/#grammardef-), which includes cookies, storage, cache, as well as execution contexts (documents, workers, etc.). We don’t differentiate between different types of site data because:
+
+ * A user identifier could be stored in any of these storage types.
+ * Clearing just a few of the types would break sites that expect different types of data to be consistent with each other.
+
+Since member sites can only add/remove themselves to/from FPSs with the consent from the owner, we look at first-party set changes as a site changing its FPS owner.
+
+If a site’s owner changed:
+
+1. If this site has no FPS owner, the site's data won't be cleared.
+    *   Pro: Avoids adoption pain when a site joins a FPS.
+    *   Con: Unclear how this lines up with user expectations about access to browsing history prior to set formation.
+2. Otherwise, clear site data of this site.
+
+Potential modification, which adds implementation complexity:
+
+3. If this site's new owner is a site that previously had the same FPS owner as the first site, the site's data won't be cleared. 
+    *   Pro: Provides graceful transitions for examples (f) and (g).
+    *   Con: Multi-stage transitions, such as (h) to (i) are unaccounted for.
+
+## Examples
+
+![FPS representation](./image/FPS_clear_site_data-representation.drawio.svg)
```

Comment:
This is a weak suggestion, since I'm not an expert on [alt text](https://webaim.org/techniques/alttext/), but I think I'd omit the alt text for all of these diagrams. Their content is just an illustration of the text below them, and screen reader users should focus on that text, rather than the diagrams.

```suggestion
![](./image/FPS_clear_site_data-representation.drawio.svg)
```